sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/http_challenge.py

```diff
@@ -40,6 +40,10 @@ def __init__(self, request_uri, challenge, response_headers=None):
         if "authorization" not in self._parameters and "authorization_uri" not in self._parameters:
             raise ValueError("Invalid challenge parameters")
 
+        authorization_uri = self.get_authorization_server()
+        # the authoritzation server URI should look something like https://login.windows.net/tenant-id
+        self.tenant_id = authorization_uri.split("/")[-1] or None
```

Comment:
Just the paranoia talking, but I would take the first segment of the path, not the last.